client/extensions/woocommerce/state/ui/shipping/classes/test/reducers.js

```diff
@@ -0,0 +1,209 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { expect, assert } from 'chai';
+import { cloneDeep } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	editShippingClass,
+	closeShippingClass,
+	addShippingClass,
+	updateShippingClassSetting,
+	saveCurrentlyOpenShippingClass,
+	removeShippingClass,
+} from '../actions';
+import {
+	updateShippingClassSuccess,
+	createShippingClassSuccess,
+	deleteShippingClassSuccess,
+} from 'woocommerce/state/sites/shipping-classes/actions';
+import reducer from '../reducer';
+
+const siteId = 123;
+const classId = 999;
+
+const initialState = {
+	editing: false,
+	editingClass: null,
+	changes: {},
+	created: [],
+	deleted: [],
+	updates: [],
+};
+
+describe( 'Shipping classes UI reducer', () => {
+	const expectedEndState = cloneDeep( initialState );
+
+	afterEach( () => {
+		// make sure the state hasn't been mutated
+		// after each test
+		expect( initialState ).to.eql( expectedEndState );
+	} );
+
+	test( 'WOOCOMMERCE_SHIPPING_CLASS_ADD', () => {
+		const action = addShippingClass( siteId );
+		const state = reducer( initialState, action );
+
+		expect( state.editing ).to.equal( true );
+		expect( state.editingClass ).to.equal( null );
+		expect( state.changes ).to.eql( {} );
+	} );
+
+	test( 'WOOCOMMERCE_SHIPPING_CLASS_EDIT', () => {
+		const action = editShippingClass( siteId, classId );
+		const state = reducer( initialState, action );
+
+		expect( state.editing ).to.equal( true );
+		expect( state.editingClass ).to.equal( classId );
+		expect( state.changes ).to.eql( {} );
+	} );
+
+	test( 'WOOCOMMERCE_SHIPPING_CLASS_CLOSE', () => {
+		const action = closeShippingClass( siteId );
+		const state = reducer( initialState, action );
+
+		expect( state.editing ).to.equal( false );
+		expect( state.editingClass ).to.equal( null );
+		expect( state.changes ).to.eql( {} );
+	} );
+
+	test( 'WOOCOMMERCE_SHIPPING_CLASS_CHANGE', () => {
+		const name = 'Shipping Class 123';
+		const action = updateShippingClassSetting( siteId, 'name', name );
+		const state = reducer( initialState, action );
+
+		expect( state.changes.name ).to.equal( name );
+	} );
+
+	test( 'WOOCOMMERCE_SHIPPING_CLASS_SAVE for a new class', () => {
+		const action = saveCurrentlyOpenShippingClass( siteId );
+		const name = 'Shipping Class 123';
+		const slug = 'shipping-class-123';
+
+		const state = reducer(
+			{
+				...initialState,
+				editing: true,
+				changes: {
+					name,
+					slug,
+				},
+			},
+			action
+		);
+
+		expect( state.editing ).to.equal( false );
+		expect( state.changes ).to.eql( {} );
+		expect( state.created ).to.have.members( [ state.created[ 0 ] ] );
```

Comment:
Does this essentially amount to asserting `length === 1`?

Reply:
It does, but it also asserts that there are particular elements in the array.

```js
expect( [ 'a', 'b' ] ).to.have.members( [ 'b', 'a' ] ); // ok
expect( [ 'a', 'b', 'c' ] ).to.have.members( [ 'b', 'a' ] ); // error
```

Doing it this way is pretty much equivalent to `.to.eql()`, but ignores the order of the elements within the array.